vmm/src/lib.rs

```diff
@@ -778,6 +782,40 @@ impl Vmm {
         self.kernel_config = Some(kernel_config);
     }
 
+    fn flush_metrics_on_request(&mut self) -> std::result::Result<VmmData, VmmActionError> {
+        if let Err(e) = self.write_metrics() {
+            if let LoggerError::NeverInitialized(s) = e {
+                return Err(VmmActionError::Logger(
+                    ErrorKind::User,
+                    LoggerConfigError::FlushMetrics(s),
+                ));
+            } else {
+                return Err(VmmActionError::Logger(
+                    ErrorKind::Internal,
+                    LoggerConfigError::FlushMetrics(e.to_string()),
+                ));
+            }
+        }
+        Ok(VmmData::Empty)
+    }
+
+    fn flush_metrics_on_event(&mut self) {
```

Comment:
nit-ish: The name of this function implies that there's an event passed as parameter. It looks like it doesn't need to be a separate function, you can just write that code when handling the epoll WriteMetrics event.